rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/NavFilter.xtend

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.views.projectexplorer
+
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IFolder
+import org.eclipse.jdt.core.ICompilationUnit
+import org.eclipse.jdt.internal.core.JarPackageFragmentRoot
+import org.eclipse.jdt.internal.core.JavaElement
+import org.eclipse.jface.viewers.Viewer
+import org.eclipse.jface.viewers.ViewerFilter
+
+/** filter all classpath containers (Java RT, JUnit) and java elements from navigator
+ * 
+ *  thus the user of the rcp sees only the files relevant to him in the project explorer
+ */
+class NavFilter extends ViewerFilter {
+	override select(Viewer viewer, Object parentElement, Object element) {
+		if (element instanceof ICompilationUnit) { // hide all java compilation units
+			return false
+		}
+		if (element instanceof IFolder) { // don't show any folders that hold generated artifacts (-gen) or maven artifacts (target)
+			if (element.projectRelativePath.segments.exists[equals("bin")||equals("target") || endsWith("-gen")]) {
```

Comment:
Do we need to check all `segements` and not just `lastSegment`? I think if the parent element is filtered out, the children won't be visible as well.